### PR TITLE
Bug 1509142 - Should not display the 'Reveal Secret' link when secret is without 'data' field

### PR DESCRIPTION
**Pull request by jhadvig**
@spadgett PTAL

### CODE REVIEW

File: app/views/browse/secret.html [hunk 1]
```diff
@@ -90,6 +90,9 @@ <h2 class="mar-top-none">
                 </div>
               </div>
             </dl>
+            <p ng-if="!secret.data" class="mar-bottom-xl">
+              There are no data in this secret.
+            </p>
```
Comment (spadgett) [inline app/views/browse/secret.html]: This is grammatically correct, but reads a little funny to me. Maybe use an empty state message?

```html
<div ng-if="!secret.data" class="empty-state-message text-center">
  <h2>No data.</h2>
  <p>This secret has no data.</p>
</div>
```

**Comment by jhadvig** [comment]
@spadgett so you are right regarding the secret without data field in the`Environment From`... there is nothing to select in the value input.

bitmap

Do we want to disable the value input in that case, or just set the placeholder to just say something like "No key" ?

**Comment by spadgett** [comment]
> Do we want to disable the value input in that case, or just set the placeholder to just say something like "No key" ?

As long as it's not broken, we might just leave it for 3.7 and come back to it in 3.8.

cc @jwforres

**Comment by jwforres** [comment]
yeah if its not throwing a JS error i wouldnt touch that code in 3.7

**Comment by jhadvig** [comment]
@spadgett disabling the btn when no `data`
no-add

**Pull request merged**